### ISSUE TITLE
Add some jitter to access token expiry

### ISSUE THREAD

**Issue by rfk** [mozilla/fxa-oauth-server#370]
When we tried to ship https://github.com/mozilla/fxa-oauth-server/pull/367 and start expiring access tokens again, we were hit by a thundering herd of Firefoxes all coming to refresh their tokens at once, and we had to roll back the deployment.  In order to get this deployed, we'll have to slowly decrease the token lifetime over a number of deploys, so that we only trigger refresh attempts from part of the population.

But there's a related issue - if we set our token expiry to be e.g. two weeks, how do we prevent a  thundering herd from re-appearing every two weeks when everyone goes to refresh their token?

A simple solution might be to add a little jitter to the token expiry, by doing something like:

```
var lifetime = config.expiration.accessToken
lifetime += lifetime * (Math.random() * 10 / 100)  // allow up to an extra 10% of expiry time
if (access_token.createdAt + lifetime < Date.now) {
  reject it, it's expired
}
```

@seanmonstar @jrgm thoughts?

**Comment by seanmonstar** [comment]
Hm, depending on how often these tokens are verified, this could end up
only delaying expiration by a couple requests, since the 'jitter' isn't
saved, so a second request 5 minutes later could generate the opposite
jitter and expire it anyways...

**Comment by rfk** [comment]
Good point.  I guess we could use the low bits of the token itself as a source of "randomness" but maybe that's pushing the hackiness factor a bit far :-)

**Comment by seanmonstar** [comment]
I was actually thinking of the exact same thing. It would be using a CSRNG number that is persisted over requests...

```js
// byte / 256 to get percent, 10% of that
var jitter = Math.floor(lifetime * buf(token)[0] / 256 * 0.1); 
```

**Comment by shane-tomlinson** [comment]
@rkf and @seanmonstar - can one of you triage this issue to the appropriate column in Waffle?

**Comment by rfk** [comment]
If we're going to do this, we should do it before @jrgm makes another attempt to ship the token-expiry stuff, so I'm moving it to "next" for discussion in planning session.

**Comment by seanmonstar** [comment]
I probably should have included a message about why I dropped it into the backlog.

We discussed this last Monday, and it felt that this didn't seem like it would provide too much benefit. 

1. The expiration duration is in the config, so it's not too difficult to deploy with the config set to `6 months`, leave it a couple days, and then deploy again with `5 months`, etc.
2. With the above expiration set, the "rush" to refresh should be spaced out a little anyways, since all Firefox clients won't try at the exact same time. Instead, they'll only notice when the browser is started up, plus any other user-initiated action. So this should mean that the clients should be refreshing at different times in the day, all depending on the user.
3. Likewise, once we've managed to get the expiration down to 2 weeks (or less! a day? yes!), the clients still will only refresh depending on actions from the user.

Therefore, we punted till data proved these assumptions wrong.
